impeller/renderer/command.h

```diff
@@ -61,7 +61,7 @@ using TextureResource = Resource<std::shared_ptr<const Texture>>;
 struct TextureAndSampler {
   SampledImageSlot slot;
   TextureResource texture;
-  std::shared_ptr<const Sampler> sampler;
+  const std::unique_ptr<const Sampler>& sampler;
```

Comment:
This is potentially a dangling pointer now.  What guarantees do we have that this reference will remain valid for the lifetime of the TextureAndSampler?

If we can be confident that it will remain valid, we should add documentation.  If we aren't so sure I think maybe we should go back to `std::shared_ptr` but keep all your work converting the arguments to references.  That would eliminate all those shared_ptr copies, except this one.

Reply:
The platform sampler libraries store the sampler in a unordered_map (which has stable references even when it reallocates) and never evicts them. This should keep the references valid for the lifetime of the program.

Reply:
> This should keep the references valid for the lifetime of the program.

The lifetime of the Context, right?  Can you add a comment for this please.

Reply:
Done: https://github.com/flutter/engine/pull/49974/files#diff-c3cd8ebc7ccac686f1868502bf7b1b710dbf93217b21f34c6ab39c84095cec5aR26

Also I deleted `Sampler::isValid` :)

Reply:
I think a comment right here at the field would be appropriate too since looking at this code peoples spider sense tingles for dangling pointers.

Reply:
This code is almost done being deleted :)